azure/packages/test/scenario-runner/src/MapTrafficRunner.ts

```diff
@@ -3,35 +3,66 @@
  * Licensed under the MIT License.
  */
 import child_process from "child_process";
+import { v4 as uuid } from "uuid";
 
 import { TypedEventEmitter } from "@fluidframework/common-utils";
+import { AzureClient } from "@fluidframework/azure-client";
+import { SharedMap } from "@fluidframework/map";
+import { PerformanceEvent } from "@fluidframework/telemetry-utils";
+import { timeoutPromise } from "@fluidframework/test-utils";
 
-import { IRunConfig, IRunner, IRunnerEvents, IRunnerStatus, RunnnerStatus } from "./interface";
-import { delay } from "./utils";
+import {
+	AzureClientConnectionConfig,
+	ContainerFactorySchema,
+	IRunConfig,
+	IRunner,
+	IRunnerEvents,
+	IRunnerStatus,
+	RunnnerStatus,
+} from "./interface";
+import {
+	AzureClientConfig,
+	createAzureClient,
+	delay,
+	getScenarioRunnerTelemetryEventMap,
+	loadInitialObjSchema,
+} from "./utils";
+import { getLogger } from "./logger";
 
-export interface AzureClientConfig {
-	type: "remote" | "local";
-	endpoint?: string;
-	key?: string;
-	tenantId?: string;
-	useSecureTokenProvider?: boolean;
-	region?: string;
-}
-
-export interface ContainerTrafficSchema {
-	initialObjects: { [key: string]: string };
-	dynamicObjects?: { [key: string]: string };
-}
+// This was originally namespaced as "DocLoader"
+const eventMap = getScenarioRunnerTelemetryEventMap("MapTraffic");
 
 export interface MapTrafficRunnerConfig {
-	connectionConfig: AzureClientConfig;
+	connectionConfig: AzureClientConnectionConfig;
 	docId: string;
-	schema: ContainerTrafficSchema;
+	schema: ContainerFactorySchema;
 	numClients: number;
 	clientStartDelayMs: number;
 	writeRatePerMin: number;
 	sharedMapKey: string;
 	totalWriteCount: number;
+	client?: AzureClient;
+}
+
+export interface MapTrafficRunnerRunConfig
```

Comment:
Curious if there is a way to simplify this distinction between MapTrafficRunnerRunConfig and MapTrafficRunnerConfig. I'll come back to this.

Reply:
I agree that most of the RunnerConfigs and RunnerRunConfigs are very similar. Would be happy to hear a way to make them share more if you have one.